### PR TITLE
Remove the pin on dask.

### PR DESCRIPTION
**Pull request by gsaluja9**
This resolves the CI failures since the release of pandas 2.0.0

### CODE REVIEW

File: aperturedb/CSVParser.py [hunk 1]
```diff
@@ -49,15 +49,15 @@ def __init__(self, filename, df=None, use_dask=False):
                 self.filename,
                 blocksize = os.path.getsize(self.filename) // (cores_used * PARTITIONS_PER_CORE))
 
-        if len(self.df) == 0:
+        # len for dask fataframe needs a client.
```
Comment (drewaogle) [inline aperturedb/CSVParser.py:52]:
```suggestion
        # len for dask dataframe needs a client.
```

File: test/conftest.py [hunk 2]
```diff
@@ -81,7 +81,8 @@ def insert_data_from_csv(in_csv_file, rec_count=-1):
                       stats=True,
                       )
         assert loader.error_counter == 0
-        return data
+        # Preserve loader so that dask manager is not auto deleted.
+        return data, loader
```
Comment (drewaogle) [inline test/conftest.py:85]: I guess this is the reason for the change above? Why do we need to preserve the dask manager?

File: aperturedb/DaskManager.py [hunk 3]
```diff
@@ -17,6 +17,22 @@
...
+        # This can be overridden by the user.
+        # Create a pool of workers.
+        # TODO: see if the same pool can be reused for multiple tasks.
+        workers = self.__num_workers if self.__num_workers != \
+            -1 else int(0.9 * mp.cpu_count())
+
+        self._cluster = LocalCluster(n_workers=workers)
+        self._cluster.shutdown_on_close = False
+        self._client = Client(self._cluster)
+        dask.config.set(scheduler="distributed")
```
Comment (drewaogle) [inline aperturedb/DaskManager.py:30]: I see you switched the scope of methods for LocalCluster and Client, what was need for the change?

**Review by drewaogle — CHANGES_REQUESTED** [review]

File: aperturedb/DaskManager.py (hunk 3 above)
Comment (gsaluja9) [reply aperturedb/DaskManager.py:30]: dask distributed , although was doing all the ingestion stuff as the norm, however it was GCing the client and the cluster immediately after the context manager it earlier was confined to. I guess the len (which we use for assertions later) relies on this client, so we keep it in reference till the time were actually done.

File: test/conftest.py (hunk 2 above)
Comment (gsaluja9) [reply test/conftest.py:85]: It encapsulates the Localcluster, and client.

**Review by luisremis — DISMISSED** [review]
thanks!

File: test/conftest.py (hunk 2 above)
Comment (drewaogle) [reply test/conftest.py:85]: Ok, sounds good, can you just add something that explains below so someone can know looking at the code later?

great detective work

**Review by drewaogle — APPROVED** [review]
Comments look great!

**Pull request merged**